relayer/pkg/starknet/chain_client.go

```diff
@@ -0,0 +1,106 @@
+package starknet
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/NethermindEth/juno/core/felt"
+	starknetrpc "github.com/NethermindEth/starknet.go/rpc"
+	gethrpc "github.com/ethereum/go-ethereum/rpc"
+)
+
+// type alias for readibility
+type FinalizedBlock = starknetrpc.Block
+
+// used to create batch requests
+type StarknetBatchBuilder interface {
+	RequestBlockByHash(h *felt.Felt) (StarknetBatchBuilder, error)
+	// RequestBlockByNumber(id uint64) (StarknetBatchBuilder, error)
+	// RequestLatestPendingBlock() (StarknetBatchBuilder, error)
+	// RequestLatestBlockHashAndNumber() (StarknetBatchBuilder, error)
+	// RequestEventsByFilter(f starknetrpc.EventFilter) (StarknetBatchBuilder, error)
+	// RequestTxReceiptByHash(h *felt.Felt) (StarknetBatchBuilder, error)
+	Build() []gethrpc.BatchElem
+}
+
+var _ StarknetBatchBuilder = (*batchBuilder)(nil)
+
+type batchBuilder struct {
+	args []gethrpc.BatchElem
+}
+
+func NewBatchBuilder() StarknetBatchBuilder {
+	return &batchBuilder{
+		args: make([]gethrpc.BatchElem, 0),
+	}
+}
+
+func (b *batchBuilder) RequestBlockByHash(h *felt.Felt) (StarknetBatchBuilder, error) {
+	b.args = append(b.args, gethrpc.BatchElem{
+		Method: "starknet_getBlockWithTxs",
+		Args: []interface{}{
+			starknetrpc.BlockID{Hash: h},
+		},
+		Result: &FinalizedBlock{},
```

Comment:
i learned the result is stored within the same BatchElem type. 

and that you need to allocate the memory for the type you expect to be returned, which is json deserialized from the RPC